tests/api/utils/test_scheduler.py

```diff
@@ -56,7 +56,10 @@ async def scheduler(db: Session) -> typing.Generator:
 
 
 call_counter: int = 0
-schedule_end_time_margin = 0.5
+
+# TODO: The margin will need to rise for each additional CPU-consuming operation added along the flow,
+#  we need to consider how to decouple in the future
+schedule_end_time_margin = 0.7
```

Comment:
I think what needs to be done here is to run notifications monitoring under a feature flag
and when running scheduler tests, simply disable it